plugins/destination_testing.go

```diff
@@ -3,243 +3,241 @@ package plugins
+			return fmt.Errorf("failed to delete stale data second time: %w", err)
+		}
 	}
 
-	if receivedResource.TableName != expectedResource.TableName {
-		return fmt.Errorf("expected table name %s, got %s", expectedResource.TableName, receivedResource.TableName)
+	resourcesRead, err = p.readAll(ctx, tables[0], sourceName)
+	if err != nil {
+		return fmt.Errorf("failed to read all second time: %w", err)
 	}
+	if len(resourcesRead) != 1 {
+		return fmt.Errorf("expected 1 resource, got %d", len(resourcesRead))
+	}
+
+	if resource2.Data.Equal(resourcesRead[0]) {
+		return fmt.Errorf("expected data to be %v, got %v", resource.Data, resourcesRead[0])
+	}
+
+	return nil
+}
+
+func (*DestinationPluginTestSuite) destinationPluginTestWriteAppend(ctx context.Context, p *DestinationPlugin, logger zerolog.Logger, spec specs.Destination) error {
+	spec.WriteMode = specs.WriteModeAppend
+	if err := p.Init(ctx, logger, spec); err != nil {
+		return fmt.Errorf("failed to init plugin: %w", err)
+	}
+	tableName := "cq_test_write_append"
+	table := testdata.TestTable(tableName)
+	syncTime := time.Now().UTC()
+	tables := []*schema.Table{
+		table,
+	}
+	if err := p.Migrate(ctx, tables); err != nil {
+		return fmt.Errorf("failed to migrate tables: %w", err)
+	}
+
+	sourceName := uuid.NewString()
+	resource := schema.DestinationResource{
+		TableName: table.Name,
+		Data:      testdata.TestData(),
+	}
+
+	if err := p.writeOne(ctx, tables, sourceName, syncTime, resource); err != nil {
+		return fmt.Errorf("failed to write one second time: %w", err)
+	}
+
+	resource = schema.DestinationResource{
+		TableName: table.Name,
+		Data:      testdata.TestData(),
+	}
+	// we dont use time.now because looks like there is some strange
+	// issue on windows machine on github actions where it returns the same thing
+	// for all calls.
+	secondSyncTime := syncTime.Add(time.Second).UTC()
+	// write second time
+	if err := p.writeOne(ctx, tables, sourceName, secondSyncTime, resource); err != nil {
+		return fmt.Errorf("failed to write one second time: %w", err)
+	}
+
+	resourcesRead, err := p.readAll(ctx, tables[0], sourceName)
+	if err != nil {
+		return fmt.Errorf("failed to read all second time: %w", err)
+	}
+
+	if len(resourcesRead) != 2 {
+		return fmt.Errorf("expected 2 resources, got %d", len(resourcesRead))
+	}
+
+	if resource.Data.Equal(resourcesRead[0]) {
+		return fmt.Errorf("expected data to be %v, got %v", resource.Data, resourcesRead[0])
+	}
+
+	if resource.Data.Equal(resourcesRead[1]) {
+		return fmt.Errorf("expected data to be %v, got %v", resource.Data, resourcesRead[1])
+	}
+
+	return nil
+}
+
+func DestinationPluginTestSuiteRunner(t *testing.T, p *DestinationPlugin, spec specs.Destination, options ...DestinationPluginTestSuiteOption) {
+	t.Helper()
+	suite := &DestinationPluginTestSuite{}
+	for _, option := range options {
+		option(suite)
+	}
+	ctx := context.Background()
+	logger := getTestLogger(t)
+
+	t.Run("TestWriteOverwrite", func(t *testing.T) {
```

Comment:
Maybe we should use a list of enums instead of multiple booleans? Then this should will be a for loop of the suites we want to run (instead of running, then skipping).

We can default to all if the list is `nil`

Reply:
Tried a different approach now.